### DISCUSSION TITLE
How to get wonderful Super / Hyper Saw sound? [solved]

### DISCUSSION THREAD

**Discussion by Opekusha** [marcel-licence/esp32_basic_synth#5] [Q&A]
How to get wonderful Super / Hyper Saw sound from your code. Please tell me what needs to be added and where? I can't do it without your help.

**Comment by marcel-licence** [comment]
I've added the unison support to generate hyper/super saw sounds. Update your code and look for 
//#define USE_UNISON

This should be replaced by
#define USE_UNISON

This will activate the detune and unison sound generation.

     Have fun! =)

**Comment by tiretrak** [comment]
and this )
static float detune       = 1.0; //0.1; /* detune parameter */
static uint8_t unison     = 7; /* additional osc per voice count */

**Reply by marcel-licence** [reply]
Yes you are right. It is also mapped to cc messages:
'#ifdef USE_UNISON
    { 0x0, 0x10, "R1", NULL, Synth_SetParam, SYNTH_PARAM_DETUNE_1},
    { 0x1, 0x10, "R2", NULL, Synth_SetParam, SYNTH_PARAM_UNISON_2},
#else'
